src/main/java/InfrastructureManager/Configuration/MasterConfigurator.java

```diff
@@ -0,0 +1,192 @@
+package InfrastructureManager.Configuration;
+
+import InfrastructureManager.*;
+import InfrastructureManager.AdvantEdge.AdvantEdgeClient;
+import InfrastructureManager.Configuration.RawData.IOConfigData;
+import InfrastructureManager.Configuration.RawData.IOPortConfigData;
+import InfrastructureManager.Configuration.RawData.MasterConfigurationData;
+import InfrastructureManager.Configuration.RawData.RunnerConfigData;
+import InfrastructureManager.MatchMaking.MatchMaker;
+import InfrastructureManager.Rest.RestInput;
+import InfrastructureManager.Rest.RestOutput;
+import InfrastructureManager.Rest.RestRunner;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Configurator class for the master, that takes the values in the configuration file
+ * object and gives the master the different elements based on that (Commands, and Runners)
+ */
+public class MasterConfigurator {
+
+    private final String CONFIG_FILE_PATH = "src/main/resources/Configuration.json";
+    private MasterConfigurationData data; //Configuration File Interface
+    private Map<String,MasterInput> inputInstances;
+    private Map<String,MasterOutput> outputInstances;
+    //TODO: consider making it a singleton
+
+    public MasterConfigurator() {
+        ObjectMapper mapper = new ObjectMapper(); //Using Jackson Functionality
+        try {
+            //Map the contents of the JSON file to a java object
+            this.data = mapper.readValue(new File(CONFIG_FILE_PATH), MasterConfigurationData.class);
+            this.inputInstances = new HashMap<>();
+            this.outputInstances = new HashMap<>();
+            //this.data.getIoData().printInfo();
+        } catch (IOException e) {
+            System.out.println("Error while reading JSON Config File");
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * Based on the commands defined in the config file, configures the CommandSet instance and returns it
+     * @return The command set for the master to use
+     */
+    public CommandSet getCommands() {
+        CommandSet.getInstance().set(data.getCommands());
+        return CommandSet.getInstance();
+    }
+
+    /**
+     * Extracts data from the configuration file to assign different input instances to their names
+     */
+    private void fillInputInstances() {
+        for (IOConfigData inputData : this.data.getIoData().getInputs()) {
+            this.inputInstances.put(inputData.getName(), getInputFromType(inputData.getType()));
+        }
+    }
+
+    /**
+     * Extracts output data from the configuration file, including ports
+     */
+    private void fillOutputInstances() {
+        for (IOConfigData outputData : this.data.getIoData().getOutputs()) {
+            if (this.inputInstances.containsKey(outputData.getName())) { //If an instance is input and output
```

Comment:
Why not the same for fillInputInstances? You don't have a problem because of the call order (fillInputInstances before fillOutputInstances), but I don't see a condition that defines the order of calling fillInputInstances and fillOutputInstances.